### ISSUE TITLE
creating preview

### ISSUE THREAD

**Issue by hiralsuvagya** [wpbakery/dev-example#1]
i am writing vc add on. almost shortcode set i am stuck with backbone view.i want to show preview of options that created for software like font,font-size,color,text,stroke,fill,shadow..
ViewTestElement is my js_view
when i create window.ViewTestElement =  vc.shortcode_view.extend({...});
and i write changeShortcodeParams() that one working but that executes before element so can't found element.

i want all the params set to preview.

thank you for all your help

**Comment by pavelthq** [comment]
@hiralsuvagya , needed more details:
1) you want to create in backend side?
2) where preview should be? In edit popup or in backend view(schema)?
3) changeShortcodeParams have context this. use console.log(this) and you will see that this.$el exists - this actually is shortcode element in dom.

**Comment by pavelthq** [comment]
Also see our changeShortcodeParams usage for MessageBox. You will see that we use it very well

**Comment by hiralsuvagya** [comment]
thank you for reply.

yes i want to show preview in backend when popup open for settings.

just same like custom heading where you can see google font preview.

i am not sure how can i add preview container also?

hope you can help with this

**Comment by pavelthq** [comment]
Google fonts it is custom parameter that have own hooks and logic in composer-atts.

look into:
vc.atts.google_fonts  parse and init methods.

Actually basic idea is on init make view that will hook on change of param and then render preview.

## 

If you want this for all params, you should use something like this:

``` javascript
vc.edit_element_block_view.on('render', function() {
     // add your logic like <button>show preview</button> and action:
     // on click button: 
   // to get current params of edit form:
 var params;
   params = _.extend( {}, vc.getDefaults( vc.edit_element_block_view.model.get( 'shortcode' ) ), vc.edit_element_block_view.getParams() );

 console.log(params);
});
```

**Comment by hiralsuvagya** [comment]
thank you!! i will try and let you know

**Comment by hiralsuvagya** [comment]
sorry but i want some event which fire after popup open.
is there anything like that? i mean so i can get preview container and set all params

**Comment by pavelthq** [comment]
@hiralsuvagya look on my comment before: vc.edit_element_block_view.on('render', function() {

**Comment by hiralsuvagya** [comment]
yes i checked that vc.atts and in composer-atts.js i can see that is created for each field

**Comment by hiralsuvagya** [comment]
but i am first stuck at how to add container for preview?

**Comment by hiralsuvagya** [comment]
should i map new param for container?

**Comment by pavelthq** [comment]
What do you mean of container? Share your code to see what you are trying to get?

**Comment by hiralsuvagya** [comment]
i have same code as https://bitbucket.org/wpbakery/extend-visual-composer-plugin-example
but with font_container and some more effect fields.

thats all working fine.

remember how google font shows preview?

google font has one div as parent and span as child to preview effect of choosen field.

so i want to add div for preview.

and when i edit my add on shortcode in backend on preview element all the effect should be already applied and on change effect should be chnage on  preview element.

**Comment by pavelthq** [comment]
So vc.atts.init should be enought to make this. - subscribe to events when your value changes and make a preview like we do in googlefonts.

Or for more simple  way - just create dependency, and in dependcy hook do same as I write - subscribe to change events and make preview on change.

**Comment by hiralsuvagya** [comment]
can you give me example of vc.atts.init and how to implement it? sorry i don't have enough knowledge of backbone

**Comment by pavelthq** [comment]
Look into our vc.atts.css_animation and vc.atts.google_fonts

**Comment by hiralsuvagya** [comment]
let me know if we can connect through skype chat.

**Comment by nandlal-tjm** [comment]
```
               window.vc.atts.tk_number = {
                    init: function ( param, $field ) {
                         console.log($field);
                        $('.tk_number_field', $field).each(function() {
                            $(this).addClass('test');
                        });
                    },
                };
```
I am adding css class for testing purpose but not showing in element. Can you please tell me how to debug this code because I can not see result in console.log
When I type in console.log vc.atts I found the tk_number
